book/src/proposals/leader-duplicate-block-slashing.md

```diff
@@ -0,0 +1,99 @@
+potential forks that the cluster has to resolve.
+
+## Overview
+
+1. Slashing Condition: If you have voted on version `S` of a slot, it is a 
+slashing condition to vote on any descendant of another version `S'` of the
+slot, unless that descendant of version `S'` has seen supermajority (> 66%)
+of the stake voting on it (See section `Slashing Condition` below for how this
+proof is to be generated).
+2. If you see two blockhashes for the same block, and you are less than 
+`2^THRESHOLD` lockout on that block, drop it. Flag the block as "duplicated".
+3. Only repair a version of the "duplicated" block if you see a `Repair Proof`
+as outlined in the `Repair Proof` section below
+
+## Primitives
+
+For a bank `B`, let `A` be the latest ancestor of `B` that has gotten 
+supermajority > 66% votes. 
+
+1) Define the function `Confirmed` to be `Confirmed(B) = A`. 
+2) Define `parent(B)` to be the parent bank of `B`.
+
+The `bank_hash` of each bank `B` is now augmented to be: 
+
+`hash(Confirmed(B).slot, state_hash(B), bank_hash(parent(B)), parent(B).slot)`.
+
+We include these components to support the proof generation described
+in the sections below.
+
+## Repair Proof
+
+Let `S'` be some version of slot `S` that a validator has already detected a
+duplicate version of and dropped and marked as "duplicate" (step 2 of the
+`Overview` section above).  A validator will only repair  `S'` if it's
+provided a proof that `S'` has been confirmed (been voted on by greater than
+66% of the stake).
+
+Call such a proof `RepairProof(S')`.
+
+This proof consists of a supermajority of validator's votes, where each vote `V`:
+
+* Contains a bank hash `H(B)` for some bank `B` such that
+`Confirmed(B).slot < S'.slot` (if a supermajority confirmed `S'`, then such a
+set must exist because there must have been some initial set of validators that
+voted on `S'` before it was confrmed). This can be confirmed for each vote if
+the prover provides `state_hash(B)`, `Confirmed(B).slot`,
+`bank_hash(parent(B))`, `parent(B).slot` such that
+
+`H(B) == hash(Confirmed(B).slot, state_hash(B), bank_hash(parent(B)), parent(B).slot)`.
+
+* Show that `V.slot` is descended from `S'` by proving the bank hash `H(B)`
+can be derived from the bank hash and slot of `S'`. This is done by providing
+the trail of bank hashes and parent slots to recreate the hash.
+
+## Slashing Condition
+
+### Goals
+The proof of slashing aims to show a validator signed two votes on two bank
+hashes `H1` and `H2` for two banks `B1` and `B2` where both:
+* Chain from two different versions of the same slot `S`.
+* `Confirmed(B1) < S.slot` and `Confirmed(B2) < S.slot`.
+
+### Contents of the Proof
+Let `S` and `S'` be two versions of some slot. A proof shows two signed votes for two 
+banks `B1` and `B2` with bank hashes `H(B1)` and `H(B2)`. The proof shows:
+s
+* `Proof of Minority`: For each of these hashes `H` and each bank `B`, show 
```

Comment:
nits: naming: Proof of Unconfirm(ed)

Reply:
Strictly speaking, the inverse of minority is majority, not supermajority (=confirmed). :p

Reply:
true, thanks!